src/clarity-angular/data/datagrid/datagrid-row-detail.ts

```diff
@@ -47,4 +52,41 @@ export class DatagridRowDetail {
     @Input("clrDgReplace") set replace(value: boolean) {
         this.expand.replace = !!value;
     }
+
+    private subscription: Subscription;
+
+    ngAfterContentInit() {
+        let columnsList = this.hideableColumnService.getColumns();
+        this.updateCellsForColumns(columnsList);
+
+        // Triggered when the Cells list changes per row-renderer
+        this.cells.changes.subscribe(( cellList ) => {
+            let columnList = this.hideableColumnService.getColumns();
+            if ( cellList.length === columnList.length ) {
+                this.updateCellsForColumns(columnList);
+            }
+        });
+
+        // Used to set things up the first time but only after all the columns are ready.
+        this.subscription = this.hideableColumnService.columnListChange.subscribe(( columnList ) => {
+            // Prevents cell updates when cols and cells array are not aligned - only seems to run on init / first time.
```

Comment:
Sorry this should be removed. It's called on all changes emitted from the service. 
Will remove.